flash/data/base_viz.py

```diff
@@ -0,0 +1,111 @@
+from contextlib import contextmanager
+from typing import Any, Dict, List, Sequence
+
+from pytorch_lightning.trainer.states import RunningStage
+from torch import Tensor
+
+from flash.core.utils import _is_overriden
+from flash.data.callback import FlashCallback
+from flash.data.process import Preprocess
+from flash.data.utils import _PREPROCESS_FUNCS, _STAGES_PREFIX
+
+
+class BaseViz(FlashCallback):
+    """
+    This class is used to profile ``Preprocess`` hook outputs and visualize the data transformations.
+    It is disabled by default.
```

Comment:
as @tchaton suggested - let's vote on the name of this functionality since for the final shape it might have a better naming like `PreprocessProfiler` or similar.

Reply:
Yes. Let's do this for another PR.